File: Facts/Parameter.java
package Facts;

import Facts.exp.MemFact;
import Facts.exp.VarFact;
import com.sun.istack.internal.Nullable;

public class Parameter {

    private VarFact name;
    private final VarFact register;
    private MemFact alias;
    private final boolean isResult;

    public Parameter(VarFact name, VarFact register, boolean isResult) {
        this.name = name;
        this.register = register;
        this.isResult = isResult;
    }

    public VarFact getName() {
        return name;
    }

    public void setName(VarFact name) {
        this.name = name;
    }

    public VarFact getRegister() {
        return register;
    }

    @Nullable
    public MemFact getAlias() {
        return alias;
    }

    public void setAlias(MemFact alias) {
        this.alias = alias;
    }

    public boolean isResult() {
        return isResult;
    }

    @Override
    public String toString() {
        return name.toString();
    }
}

File: src/main/scala/astnodes/Parameter.scala
package astnodes

trait Parameter(name: Register, register: Register) {
  override def toString: String = {
    name.toString + ": bv" + name.size.get
  }
}

case class OutParameter(name: Register, register: Register) extends Parameter(name, register) {
  override def toString: String = {
    String.format("%s: bv%d, Gamma_%s: bool", name, name.size.get, name)
  }
}

case class InParameter(name: Register, register: Register) extends Parameter(name, register) {
  /*  
  var alias: MemLoad = null

  def getAlias: MemLoad = {
    alias
  }

  def setAlias(alias: MemLoad): Unit = {
    this.alias = alias
  }
  */
}